subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractAndroidSantaTrackerSmokeTest.groovy

```diff
@@ -58,10 +60,23 @@ class AbstractAndroidSantaTrackerSmokeTest extends AbstractSmokeTest {
     }
 
     protected GradleRunner runnerForLocation(File projectDir, String agpVersion, String... tasks) {
-        def runner = runner(*[["-DagpVersion=$agpVersion", "-DkotlinVersion=${TestedVersions.kotlin.latest()}"], tasks].flatten())
+        def runner = runner(*[["-DagpVersion=$agpVersion", "-DkotlinVersion=${TestedVersions.kotlin.latest()}", "--stacktrace"], tasks].flatten())
             .withProjectDir(projectDir)
             .withTestKitDir(homeDir)
-            .forwardOutput()
+            .forwardOutput() as DefaultGradleRunner
+        if (JavaVersion.current().isJava9Compatible()) {
+            runner.withJvmArguments(
+                "-Xmx8g", "-XX:MaxMetaspaceSize=1024m", "-XX:+HeapDumpOnOutOfMemoryError",
```

Comment:
Are these memory limits intentional?

Reply:
Yes.
I copied them from here https://github.com/gradle/gradle/blob/58dff8ed54258e9af71a6d4df83f32a8f39e2558/subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy#L241 as `withJvmArguments` overwrites the previously set args and this particular test looks quite massive.